### PR TITLE
Support collect ZGC memory pool metrics

### PR DESCRIPTION
**Pull request by Z-Beatles**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

for UI setting change, see [skywalking-java #622](https://github.com/apache/skywalking-java/pull/622)

### CODE REVIEW

**Comment by Z-Beatles** [comment]
image

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/general/general-instance.json [hunk 1]
```diff
@@ -326,7 +326,88 @@
...
                 {
-                  "x": 8,
+                  "x": 16,
+                  "y": 0,
+                  "w": 8,
+                  "h": 13,
+                  "i": "5",
+                  "type": "Widget",
+                  "widget": {
+                    "title": "JVM Memory Detail (MB)"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/general/general-instance.json:336]: Are you adding a new widget?
This name doesn't seem diff from other GC model.

File: oap-server/server-starter/src/main/resources/oal/java-agent.oal [hunk 2]
```diff
@@ -22,6 +22,17 @@ instance_jvm_memory_heap = from(ServiceInstanceJVMMemory.used).filter(heapStatus
 instance_jvm_memory_noheap = from(ServiceInstanceJVMMemory.used).filter(heapStatus == false).longAvg();
 instance_jvm_memory_heap_max = from(ServiceInstanceJVMMemory.max).filter(heapStatus == true).longAvg();
 instance_jvm_memory_noheap_max = from(ServiceInstanceJVMMemory.max).filter(heapStatus == false).longAvg();
+instance_jvm_memory_pool_code_cache = from(ServiceInstanceJVMMemoryPool.used).filter(poolType == MemoryPoolType.CODE_CACHE_USAGE).longAvg();
+instance_jvm_memory_pool_newgen = from(ServiceInstanceJVMMemoryPool.used).filter(poolType == MemoryPoolType.NEWGEN_USAGE).longAvg();
+instance_jvm_memory_pool_oldgen = from(ServiceInstanceJVMMemoryPool.used).filter(poolType == MemoryPoolType.OLDGEN_USAGE).longAvg();
+instance_jvm_memory_pool_survivor = from(ServiceInstanceJVMMemoryPool.used).filter(poolType == MemoryPoolType.SURVIVOR_USAGE).longAvg();
+instance_jvm_memory_pool_permgen = from(ServiceInstanceJVMMemoryPool.used).filter(poolType == MemoryPoolType.PERMGEN_USAGE).longAvg();
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/oal/java-agent.oal:29]: Why these? All thses are not new metrics.

I am not following.

**Review by wu-sheng — COMMENTED** [review]
Please share the screenshots about what is actually the diff.

**Comment by Z-Beatles** [comment]
### changes:
1. Add a new widget `JVM Memory Detail (MB)` to show memory pool detail.
2. Support collect and show ZGC memory pool metrics.

### before:
image

### after:
 Case1: app start up use default G1 gc. `-XX:+UseG1GC`
image

Case2: app start up use ZGC. `-XX:+UseZGC`
image

**Comment by wu-sheng** [comment]
Please make the submodule correctly, pointing to the master branch commit ID.

**Comment by wu-sheng** [comment]
Please make the submodule correctly, pointing to the master branch commit ID.

**Comment by wu-sheng** [comment]
As you have updated UI, I think there is UI side commit gap too. Please update the UI part change logs accordingly.

**Comment by wu-sheng** [comment]
Still missing one. 

image

Please check from git history https://github.com/apache/skywalking-booster-ui/commits/main

**Comment by Z-Beatles** [comment]
> Still missing one.
> 
> image
> Please check from git history https://github.com/apache/skywalking-booster-ui/commits/main
Which one missing? There are no changes in the [skywalking-booster-ui](https://github.com/apache/skywalking-booster-ui).

**Comment by wu-sheng** [comment]
The one I was showing above. It wasn't synced, but you are syncing sub module now, so, you need to add this to change log as well.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**